compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/TypeChecker.java

```diff
@@ -7606,6 +7608,13 @@ private BType checkMappingField(RecordLiteralNode.RecordField field, BType mappi
                             }
                         }
                     }
+                    if (!spreadRecordType.sealed && (mappingRecordType.sealed ||
+                            !types.isAssignable(spreadRecordType.restFieldType, mappingRecordType.restFieldType))) {
+                        dlog.error(spreadExpr.pos,
+                                DiagnosticErrorCode.INVALID_SPREAD_OP_TO_CREATE_CLOSED_RECORD_FROM_OPEN_RECORD,
+                                spreadExprType);
+                        errored = true;
```

Comment:
It's the rest descriptor, not a binding pattern. May also be misleading when used against an inclusive record type descriptor?

Reply:
In that case can we have two separate errors.
For example for this code we can give the errror, `invalid usage of spread field in record construction; mismatching record rest type descriptor`
```ballerina
type Student record {|
    string name;
    int age;
    int...;
|};

type Employee record {|
    string name;
    int age;
    string ...;
|};

public function main() {
    Employee emp = {name: "A", age: 1};
    Student _ = {...emp};
}
```

And for this we can give the errror, `invalid usage of spread field to construct a record with rest field 'int' from a inclusive record type descriptor`
```ballerina
type Student record {|
    string name;
    int age;
    int...;
|};

type Employee record {
    string name;
    int age;
};

public function main(testCommitOrRollbackInIfWithoutElseResult); {
    Employee emp = {name: "A", age: 1};
    Student _ = {...emp};
}

```

Reply:
We need two errors when trying to construct an open vs closed record IMO.

Maybe 

```
invalid usage of spread field with open record of type 'Employee' that may have rest fields of type 'int' to construct a record that allows only 'string' rest fields
```

(this can be used in the inclusive case too)

and

```
invalid usage of spread field with open record of type 'Employee' that may have rest fields to construct a closed record
```